silkrpc/consensus/ethash.cpp

```diff
@@ -16,15 +16,43 @@
 
 #include "ethash.hpp"
 
+#include <silkworm/core/silkworm/chain/protocol_param.hpp>
+#include <silkrpc/json/types.hpp>
+
 namespace silkrpc::ethash {
 
 BlockReward compute_reward(const ChainConfig& config, const silkworm::Block& block) {
-    return BlockReward{};
+    auto chain_config = silkworm::ChainConfig::from_json(config.config).value();
```

Comment:
`silkworm::ChainConfig::from_json` returns `std::optional<silkworm::ChainConfig>`, so you are missing a runtime check: if no chain configuration is returned, you should handle the error.
Moreover, once obtained a surely valid `chain_config`, it can be declared `const`.

Reply:
Right.
I didn't find a specific exception for this error that seems an invalid or bad configuration. So my choice was to throw a generic std::runtime_error.